spec/features/user/dashboard_spec.rb

```diff
@@ -195,10 +195,10 @@
       end
       UserNpcModel.create!(npc_model_id: NpcModel.all[0].id, user_id: @user_1.id)
       NpcModel.all[0].update(is_saved: 1)
-      UserNpcModel.create!(npc_model_id: NpcModel.all[1].id, user_id: @user_1.id)
-      NpcModel.all[1].update(is_saved: 1)
-      UserNpcModel.create!(npc_model_id: NpcModel.all[2].id, user_id: @user_2.id)
-      NpcModel.all[2].update(is_saved: 1)
+      UserNpcModel.create!(npc_model_id: NpcModel.all[0].id, user_id: @user_1.id)
+      NpcModel.all[0].update(is_saved: 1)
+      UserNpcModel.create!(npc_model_id: NpcModel.all[0].id, user_id: @user_2.id)
+      NpcModel.all[0].update(is_saved: 1)
```

Comment:
This logic was just not doing the correct thing. Every time it would update the npc, that npc would be shifted to the end, so not all of them were being updated.  This was appears to be working correctly